tests/testthat/helper-memory.R

```diff
@@ -0,0 +1,11 @@
+maybe_referenced <- function(x) {
+  .Call(vctrs_maybe_referenced, x)
```

Comment:
This will always be referenced because the input is binded in the execution env of the closure wrapper. To work around this you need to defuse `x` with `ensym()` and pass the caller env along with the defused symbol to the native routine, and then use `r_env_get()`.